### PR TITLE
+ call onCancel when pressing esc key

### PR DESCRIPTION
**Pull request by elios264**
I think this is it, 

BUT, I couldn't test in on my computer since I use windows, and I can't get it to compile.

fix issue  #5203

### CODE REVIEW

File: components/modal/Modal.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export interface ModalFuncProps {
   iconType?: string;
 }
 export type ModalFunc = (props: ModalFuncProps) => {
-  destroy: () => void
+  destroy: (e) => void
```
Comment (benjycui) [inline components/modal/Modal.tsx]: ref: https://github.com/ant-design/ant-design/issues/2110

Here is a problem, if developers call destroy, should he pass an event to it?

Comment (benjycui) [inline components/modal/Modal.tsx]: cc @afc163 @RaoHai

File: components/modal/confirm.tsx [hunk 2]
```diff
@@ -30,10 +30,11 @@ export default function confirm(config) {
     (props.okCancel ? runtimeLocale.okText : runtimeLocale.justOkText);
   props.cancelText = props.cancelText || runtimeLocale.cancelText;
 
-  function close() {
+  function close(e) {
     const unmountResult = ReactDOM.unmountComponentAtNode(div);
     if (unmountResult && div.parentNode) {
       div.parentNode.removeChild(div);
+      props.onCancel(e);
```
Comment (afc163) [inline components/modal/confirm.tsx]: Put `props.onCancel` outside `if`.

Comment (afc163) [inline components/modal/confirm.tsx]:
```jsx
function close(...args) {
  ...

  props.onCancel(...args);
}
```

prevent problem @benjycui describes

**Review by afc163 — CHANGES_REQUESTED** [review]
check comments

**Comment by afc163** [comment]
@elios264

**Review by benjycui — APPROVED** [review]

**Pull request merged**